lte/gateway/c/session_manager/BUILD.bazel

```diff
@@ -0,0 +1,494 @@
+        "//orc8r/gateway/c/common/logging",
+        "//orc8r/protos:redis_cpp_proto",
+        "@com_github_google_glog//:glog",
+        "@cpp_redis",
+        "@yaml-cpp//:yaml-cpp",
+    ],
+)
+
+cc_library(
+    name = "session_reporter",
+    srcs = ["SessionReporter.cpp"],
+    hdrs = ["SessionReporter.h"],
+    # TODO(@themarwhal): Migrate to using full path for includes - GH8494
+    strip_include_prefix = "/lte/gateway/c/session_manager",
+    deps = [
+        ":grpc_magma_utils",
+        "//lte/gateway/c:folly",
+        "//lte/protos:session_manager_cpp_grpc",
+        "//orc8r/gateway/c/common/async_grpc:async_grpc_receiver",
+        "//orc8r/gateway/c/common/service_registry",
+        "@com_github_google_glog//:glog",
+    ],
+)
+
+cc_library(
+    name = "stored_state",
+    srcs = ["StoredState.cpp"],
+    hdrs = ["StoredState.h"],
+    # TODO(@themarwhal): Migrate to using full path for includes - GH8494
+    strip_include_prefix = "/lte/gateway/c/session_manager",
+    deps = [
+        ":credit_key",
+        ":types",
+        "//lte/gateway/c:folly",
+        "//lte/protos:pipelined_cpp_grpc",
+        "//lte/protos:session_manager_cpp_grpc",
+        "//orc8r/gateway/c/common/logging",
+        "@com_github_google_glog//:glog",
+    ],
+)
+
+cc_library(
+    name = "rule_store",
+    srcs = ["RuleStore.cpp"],
+    hdrs = ["RuleStore.h"],
+    # TODO(@themarwhal): Migrate to using full path for includes - GH8494
+    strip_include_prefix = "/lte/gateway/c/session_manager",
+    deps = [
+        ":credit_key",
+        "//lte/protos:pipelined_cpp_grpc",
+        "//orc8r/gateway/c/common/async_grpc:async_grpc_receiver",
+        "//orc8r/gateway/c/common/service_registry",
+        "@com_github_google_glog//:glog",
+    ],
+)
+
+cc_library(
+    name = "session_credit",
+    srcs = [
+        "ChargingGrant.cpp",
+        "EnumToString.cpp",
+        "SessionCredit.cpp",
+    ],
+    hdrs = [
+        "ChargingGrant.h",
+        "EnumToString.h",
+        "Monitor.h",
+        "SessionCredit.h",
+    ],
+    # TODO(@themarwhal): Migrate to using full path for includes - GH8494
+    strip_include_prefix = "/lte/gateway/c/session_manager",
+    deps = [
+        ":diameter_codes",
+        ":service_action",
+        ":stored_state",
+        ":utilities",
+        "//lte/protos:abort_session_cpp_proto",
+        "//lte/protos:pipelined_cpp_grpc",
+        "//lte/protos:session_manager_cpp_grpc",
+    ],
+)
+
+cc_library(
+    name = "shard_tracker",
+    srcs = ["ShardTracker.cpp"],
+    hdrs = ["ShardTracker.h"],
+    # TODO(@themarwhal): Migrate to using full path for includes - GH8494
+    strip_include_prefix = "/lte/gateway/c/session_manager",
+)
+
+cc_library(
+    name = "session_state",
+    srcs = ["SessionState.cpp"],
+    hdrs = ["SessionState.h"],
+    # TODO(@themarwhal): Migrate to using full path for includes - GH8494
+    strip_include_prefix = "/lte/gateway/c/session_manager",
```

Comment:
do these strip_include_prefix args do something?

and should we be preferring full import paths? if so, do we need a strip/includes_prefix or will bazel do the right thing and use paths with workspace dir as the base?

Reply:
Yes, so since we don't use full import paths, I've had to add the strip include prefix for now. GH8494 tracks the work for converting paths to be full path. If we fix up all the paths, we shouldn't need any special configuration here and everywhere else.